pkg/storage/dimension/dimension.go

```diff
@@ -144,3 +144,19 @@ func Intersection(input ...*Dimension) []key {
 		}
 	}
 }
+
+func Union(input ...*Dimension) []key {
```

Comment:
As it's implemented right now, this function will return a lot of duplicates I think. We should optimize it to take advantage of the fact that the input dimensions are all sorted arrays, similarly to how `Intersection` is implemented.